### PR TITLE
[HHH-9876] Ability to filter objects from Database for schema tooling

### PR DESCRIPTION
**Pull request by marcus-nl**
Including unit test

### CODE REVIEW

**Comment by sebersole** [comment]
This is quite out of date with master.  Please rebase and incorporate the upstream changes

**Comment by marcus-nl** [comment]
Done (but using merge rather than rebase)

**Comment by sebersole** [comment]
In the future we prefer rebase.  But no worries, when I rebase git will remove your merge commits.  I will take a look next week (hopefully Monday)

**Comment by marcus-nl** [comment]
I've updated this PR with the changes as discussed in https://hibernate.atlassian.net/browse/HHH-9876

File: hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java [hunk 1]
```diff
@@ -755,8 +755,26 @@
...
+	 * @since 5.1
+	 */
+	String SCHEMA_DROP_FILTER = "hibernate.schema.drop.filter";
 
+	/**
+	 * Used to specify the {@link org.hibernate.tool.schema.spi.SchemaFilter} to use for migrating the database schema.
+	 * 
+	 * @since 5.1
+	 */
+	String SCHEMA_MIGRATE_FILTER = "hibernate.schema.migrate.filter";
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java]: If we need support for varied filtering for each I think I'd prefer one contract (and one setting) that gives access to the appropriate filter.  Something like:

````
interface SchemaFilterProvider {
    SchemaFilter getCreateFilter();
    SchemaFilter getDropFilter();
    SchemaFilter getMigrateFilter();
    SchemaFilter getValidateFilter();
}
````

**Comment by marcus-nl** [comment]
Done.

**Comment by sebersole** [comment]
This work references an interface named SchemaFilterProvider and a class named DefaultSchemaFilterProvider, but the PR does not seem to include them.  Am I just missing them?

**Comment by marcus-nl** [comment]
Sorry, here they are.

**Comment by sebersole** [comment]
Applied upstream.  Thanks Marcus!

**Pull request closed without merging**